idaes/generic_models/unit_models/column_models/column.py

```diff
@@ -0,0 +1,1144 @@
+            balance_type=EnergyBalanceType.enthalpyTotal,
+            has_heat_transfer=True)
+
+        self.vapor_phase.add_momentum_balances(
+            balance_type=MomentumBalanceType.pressureTotal,
+            has_pressure_change=self.config.vapor_side.has_pressure_change)
+
+        self.vapor_phase.apply_transformation()
+
+    # ==========================================================================
+        """ Build Control volume 1D for liquid phase and
+            populate liquid control volume
+
+        """
+        self.liquid_phase = ControlVolume1DBlock(default={
+            "transformation_method": self.config.transformation_method,
+            "transformation_scheme":
+                self.config.liquid_side.transformation_scheme,
+            "finite_elements": self.config.finite_elements,
+            "collocation_points": self.config.collocation_points,
+            "dynamic": self.config.dynamic,
+            "has_holdup": self.config.has_holdup,
+            "area_definition": DistributedVars.variant,
+            "property_package": self.config.liquid_side.property_package,
+            "property_package_args":
+                self.config.liquid_side.property_package_args})
+
+        self.liquid_phase.add_geometry(flow_direction=set_direction_liquid,
+                                       length_domain_set=self.config.
+                                       length_domain_set)
+
+        self.liquid_phase.add_state_blocks(
+            information_flow=set_direction_liquid,
+            has_phase_equilibrium=False)
+
+        self.liquid_phase.add_material_balances(
+            balance_type=MaterialBalanceType.componentTotal,
+            has_phase_equilibrium=False,
+            has_mass_transfer=True)
+
+        self.liquid_phase.add_energy_balances(
+            balance_type=EnergyBalanceType.enthalpyTotal,
+            has_heat_transfer=True)
+
+        self.liquid_phase.apply_transformation()
+
+        # Add Ports for vapor side
+        self.add_inlet_port(name="vapor_inlet", block=self.vapor_phase)
+        self.add_outlet_port(name="vapor_outlet", block=self.vapor_phase)
+
+        # Add Ports for liquid side
+        self.add_inlet_port(name="liquid_inlet", block=self.liquid_phase)
+        self.add_outlet_port(name="liquid_outlet", block=self.liquid_phase)
+
+    # ==========================================================================
+        """ Add performace equation method"""
+        self._make_performance()
+
+    def _make_performance(self):
+        """
+        Constraints for unit model.
+
+        Args: None
+
+        Returns: None
+
+        """
+
+        # ======================================================================
+        # Custom Sets
+        vap_comp = self.config.vapor_side.property_package.component_list
+        liq_comp = self.config.liquid_side.property_package.component_list
+        equilibrium_comp = vap_comp & liq_comp
+        solvent_comp_list = \
```

Comment:
Do we need the solvent and solute sets in the generic model? I would think that the general model would just define mass transfer for all common components, and the user then needs to define these.